addons/udes_stock/models/container_type.py

```diff
@@ -0,0 +1,59 @@
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class ContainerType(models.Model):
+    _name = "container.type"
+    _description = "Container Type"
+    _inherit = "mail.thread"
+
+    active = fields.Boolean(default=True, string="Active", tracking=True)
+    name = fields.Char(
+        string="Name",
+        required=True,
+        tracking=True,
+        help="The name of the container type, e.g Medium Box 1.",
+    )
+    length = fields.Float(
+        string="Length (mm)",
+        required=True,
+        tracking=True,
+        help="The length of the container type in mm.",
+    )
+    width = fields.Float(
+        string="Width (mm)",
+        required=True,
+        tracking=True,
+        help="The width of the container type in mm.",
+    )
+    height = fields.Float(
+        string="Height (mm)",
+        required=True,
+        tracking=True,
+        help="The height of the container type in mm.",
+    )
+    weight = fields.Float(
+        string="Weight (kg)",
+        required=True,
+        tracking=True,
+        help="The weight of the container type in kg.",
+    )
+
+    _sql_constraints = [
+        (
+            "name_uniq",
+            "unique (name)",
+            "A container type with that name already exists.",
+        ),
+    ]
+
+    @api.constrains("length", "width", "height", "weight")
+    def _check_dimensions_and_weight(self):
+        for record in self:
+            if record.length <= 0 or record.width <= 0 or record.height <= 0 or record.weight < 0:
```

Comment:
NIT: To many or-s, maybe can be improved a little the if control, i would use if any in self or filter the records from self with values less than 0 maybe